src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -375,54 +361,85 @@ private bool PrunePackages(PublishRoot root)
                 foreach (var library in target.Libraries)
                 {
                     var packagesDir = resolver.GetInstallPath(library.Name, library.Version);
-                    var manifest = resolver.GetManifestFilePath(library.Name, library.Version);
-
-                    keep.Add(manifest);
 
                     foreach (var path in library.RuntimeAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
                     }
 
                     foreach (var path in library.CompileTimeAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
                     }
 
                     foreach (var path in library.NativeLibraries)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
                     }
 
                     foreach (var path in library.ResourceAssemblies)
                     {
-                        keep.Add(CombinePath(packagesDir, path));
+                        keep.Add(Path.Combine(packagesDir, Path.GetDirectoryName(path)));
                     }
+                }
+            }
 
-                    foreach (var specialFolder in specialFolders)
-                    {
-                        var specialFolderPath = CombinePath(packagesDir, specialFolder);
+            var frameworkSpecificContents = new List<string>
+            {
+                "lib",
+                "ref"
+            };
 
-                        if (!Directory.Exists(specialFolderPath))
-                        {
-                            continue;
-                        }
+            var extensionsToPrune = new List<string>
+            {
+                "xml"
```

Comment:
I'm probably missing something here but I couldn't figure a way of using the lock file contents.

1. The lock file in the publish output only lists the files required by the frameworks specified during publish in the `targets` section. There is a `libraries` section which lists all the files in the package. However, there are files there that are used but not listed in `targets` like `runtimes/*` so I can't just prune the difference between the two sections.

2. I can check the difference between the `targets` section of the lockfiles in the project directory and the publish output. This has two problems. One, if you publish for the same frameworks you stored with, which I assume is the most common usage, nothing will be pruned. Two, even if you publish for one of the many frameworks you restored with. Pruning will miss such a large number of files, it'll almost become unnecessary. 

For example, say you restored the `System.Text.Encoding` package for `dnx451` and `dnxcore50` and you publish for `dnx451` only. The difference between the lockfiles' `targets` section will be only `lib/dnxcore50/*.dll`. Not only will this miss tons of unused frameworks like `lib/(MonoAndroid10|xamarinios10|net45|...)/*.dll`, it will also miss files that never appear in the `targets` section like `xml`, `sha512`, `nupkg`.

Hence I tried a compromise where instead of white listing a set of files and delete the rest, we black list a smaller set of files to prune that we are confident that won't be used.

But I'm assuming you still prefer option 2?